### DISCUSSION TITLE
Option to use the Model name as the Filename

### DISCUSSION THREAD

**Discussion by Terroosh** [BlafKing/sd-civitai-browser-plus#247] [Ideas]
The filenames for models uploaded to civitai often have nothing to do with the actual model name. A neat feature would be an option in settings to replace the filename with the name of the model.

![firefox_JTIWVEMhK4](https://github.com/BlafKing/sd-civitai-browser-plus/assets/55752005/492dc574-6aa1-4ce5-af62-c03e09d94457)

In this example the model would be named as Eye size adjuster.safetensors instead of yjdxtz.safetensors

**Comment by hakaserver** [comment]
You might not know this, but the 'Model filename' field is editable! This means you can always click on it and edit the name of the file on those cases. After editing the name, all acompanying files like images and .json files will also have the edited name! ✌️

**Comment by Terroosh** [comment]
> You might not know this, but the 'Model filename' field is editable! This means you can always click on it and edit the name of the file on those cases. After editing the name, all acompanying files like images and .json files will also have the edited name! ✌️

I am aware of that however doing that for every model downloaded can become quite tedious.

**Comment by BlafKing** [comment]
Sounds good!

Only one thing to consider is that the filename is used as a fallback to detect models being installed if the sha256 hash fails for whatever reason, so I'll probably include the original filename in the matching json or something similair.

Regardless, I'll add it to the to-do list! 👍🏻
